flows/inputs/msg.go

```diff
@@ -87,12 +87,15 @@ func ReadMsgInput(env flows.SessionEnvironment, envelope *utils.TypedEnvelope) (
 	return &input, nil
 }
 
-func (r *MsgInput) MarshalJSON() ([]byte, error) {
-	envelope := msgInputEnvelope{
-		baseInputEnvelope: baseInputEnvelope{ChannelUUID: r.Channel().UUID(), CreatedOn: r.CreatedOn()},
-		URN:               r.urn,
-		Text:              r.text,
+func (i *MsgInput) MarshalJSON() ([]byte, error) {
+	var envelope msgInputEnvelope
+
+	if i.Channel() != nil {
```

Comment:
This is an unrelated fix to stop marshalling of run.input blowing up if input is channel-less